File: scripts/452findMinArrowShots.py
from typing import List
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x:x[0])
        prev_end = points[0][1]
        ans = 1
        for i in range(1, len(points)):
            if points[i][0] <= prev_end:
                prev_end = min(points[i][1], prev_end)
            else:
                ans += 1
                prev_end = points[i][1]
        return ans

points = [[1,2],[2,3],[3,4],[4,5]]
print(f"result: {Solution().findMinArrowShots(points)}")

File: src/452findMinArrowShots.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int prev_end = points[0][1], ans = 1;
        for(int i = 1; i < points.size(); i++){
            if(points[i][0] <= prev_end){
                prev_end = min(points[i][1], prev_end);
            }else{
                ans++;
                prev_end = points[i][1];
            }
        }
        return ans;
    }
};

int main(){
    Solution su;
    vector<vector<int>>points = {{1, 2}, {2, 3}, {3, 4}, {4, 5}};
    cout << "result :" << su.findMinArrowShots(points) << endl;
    return 0;
}